filters/glitch/src/GlitchFilter.js

```diff
@@ -33,37 +38,91 @@ export default class GlitchFilter extends PIXI.Filter {
             direction: 0,
             fillMode: 0,
             average: false,
-            seed: 0.5,
+            seed: 0,
             red: [0, 0],
             green: [0, 0],
             blue: [0, 0],
-            minSliceWidth: 8,
-            displacementMapSize: 512,
-            displacementMap: null,
+            minSize: 8,
+            sampleSize: 512,
         }, options);
 
-        this.offset = options.offset;
         this.direction = options.direction;
+        this.red = options.red;
+        this.green = options.green;
+        this.blue = options.blue;
 
+        /**
+         * The maximum offset value for each of the slices.
+         *
+         * @member {number}
+         */
+        this.offset = options.offset;
+
+        /**
+         * The fill mode of the space after the offset.
+         *
+         * @member {number}
+         */
         this.fillMode = options.fillMode;
+
+        /**
+         * average
+         *
+         * @member {boolean}
+         */
         this.average = options.average;
+
+        /**
+         * A seed value for randomizing color offset. Animating
+         * this value to `Math.random()` produces a twitching effect.
+         *
+         * @member {number}
+         */
         this.seed = options.seed;
-        this.red = options.red;
-        this.green = options.green;
-        this.blue = options.blue;
-        this.minSliceWidth = options.minSliceWidth;
-        this.displacementMapSize = options.displacementMapSize;
-        this.displacementMap = options.displacementMap;
-
-        if (!this.displacementMap) {
-            this.displacementMapCanvas = document.createElement('canvas');
-            this.displacementMapCanvas.width = 8;
-            this.displacementMapCanvas.height = this.displacementMapSize;
-            this.displacementMap = PIXI.Texture.fromCanvas(this.displacementMapCanvas, PIXI.SCALE_MODES.NEAREST);
-
-            this._slices = 0;
-            this.slices = options.slices;
-        }
+
+        /**
+         * Minimum size of slices as a portion of the `sampleSize`
+         *
+         * @member {number}
+         */
+        this.minSize = options.minSize;
+
+        /**
+         * Height of the displacement map canvas.
+         *
+         * @member {number}
+         * @readonly
+         */
+        this.sampleSize = options.sampleSize;
+
+        /**
+         * Internally generated canvas.
+         *
+         * @member {HTMLCanvasElement} _canvas
+         * @private
+         */
+        this._canvas = document.createElement('canvas');
+        this._canvas.width = 8;
```

Comment:
I think    4 is  enough & safe.  (2 is not safe)
in the shader , will read the middle of width .